src/shared/uq/uq_url_utils.py

```diff
@@ -0,0 +1,21 @@
+import re
+
+
+class UqProductCodeParser:
+    """
+    A class with can parse product code from UNIQLO url
+    """
+
+    def __init__(self):
+        self.pattern = re.compile(r"u[0-9]+")
```

Comment:
```suggestion
    pattern = re.compile(r"u[0-9]+")
```
We can remove the constructor here since no argument is passed in.